crates/router/src/core/webhooks/utils.rs

```diff
@@ -154,3 +158,98 @@ pub(crate) fn get_idempotent_event_id(
 pub(crate) fn generate_event_id() -> String {
     common_utils::generate_time_ordered_id("evt")
 }
+
+pub async fn perform_redis_lock<A>(
```

Comment:
Nit: Can keep the visibility of the `perform_redis_lock()` and `free_redis_lock()` functions as `pub(super)`, so that they're only visible within the `webhooks` module.